tests/infra_test.go

```diff
@@ -58,6 +58,53 @@ var _ = Describe("Infrastructure", func() {
 		var pod *k8sv1.Pod
 		var metricsURL string
 
+		// start a VMI, wait for it to run and return the node it runs on
+		startVMI := func(vmi *v1.VirtualMachineInstance) string {
```

Comment:
what's the benefit of defining those functions inline instead of having them free functions like before? Could you please amend the commit message to describe it?

Reply:
Well, the benefit is that it can access the "state" of the virtclient, the pod and the vmi directly and it also limits their scope to the prometheus tests

Reply:
I'm not a fan of this, it seems to me the benefits aren't too clear - and this feels... too dynamic. But it', not fighting this too much.